### PR TITLE
feat(latin validator): exclude c0 latin chars

### PR DESCRIPTION
**Pull request by lalitaCF**
ref #https://github.com/crazyfactory/shop/issues/6216

### CODE REVIEW

**Comment by turbo8p** [comment]
The emoji character seems to be out of the context of LatinValidator.
It might be better if we create AddressLineValidator and then validate

1. latin
2. emoji

Any future validation about the AddressLineValidator can be added to this class without breaking its context and meaning.

**Review by z3niths — APPROVED** [review]

File: tests/unit/LatinCharValidatorTest.php [hunk 1]
```diff
@@ -54,5 +54,9 @@ public function testIsValid()
         $this->assertFalse(LatinCharValidator::isValid('الْأَبْجَدِيَّة الْعَرَبِيَّة‎'));
         // Russian chars
         $this->assertFalse(LatinCharValidator::isValid('Меня зовут Мандли'));
+        // moji
+        $this->assertFalse(LatinCharValidator::isValid('Hello World 😀'));
```
Comment (z3niths) [inline tests/unit/LatinCharValidatorTest.php:58]: isValid ?

Comment (z3niths) [reply tests/unit/LatinCharValidatorTest.php:58]: And we can also split into 2 validator so we can use it in the different context separaltly if we want

**Comment by lalitaCF** [comment]
� == U+0007

first, I thought it was emoji but actually, it is in basic latin.

image

**Review by turbo8p — APPROVED** [review]

**Comment by turbo8p** [comment]
Nice addition for the doc comment.

**Review by turbo8p — CHANGES_REQUESTED** [review]
seems like not all characters in the table are detected by regex.
please double check again.

I replace the string in the added test with these character and test failed.
image

File: src/LatinCharValidator.php [hunk 2]
```diff
@@ -4,8 +4,14 @@
...
+    /**
+     * exclude C0 controls (basic latin): https://unicode-table.com/en/blocks/basic-latin/
+     * @param  string  $txt
+     *
+     * @return bool
+     */
     public static function isValid(string $txt): bool
     {
-        return empty($txt) || preg_match('/^[\p{Latin}\p{Common}]+$/u', $txt) > 0;
+        return empty($txt) || (preg_match('/^[\p{Latin}\p{Common}]+$/u', $txt) > 0 && preg_match('/[\x{0000}-\x{001F}]/u', $txt) == 0 );
```
Comment (turbo8p) [inline src/LatinCharValidator.php:15]: Isn't is supposed to be `!preg_match(addition latin pattern)` ? `!` is missing.

Comment (turbo8p) [inline src/LatinCharValidator.php:15]: isn't is supposed to be `!preg_match(c0 pattern)` ? `!` is missing.

**Comment by turbo8p** [comment]
alright. i was confused at first because copy the character directly from the web and then paste in IDE isn't accurate. The unicode character has been manipulated when doing that.

In order to get the correct character from the unicode table, I need to:
- enter this website https://www.online-toolz.com/tools/text-unicode-entities-convertor.php
- Enter unicode code and then convert
- Copy from the result box
image

**Review by turbo8p — APPROVED** [review]

**Pull request merged**